### PR TITLE
exclude non-employees from message list

### PR DESCRIPTION
**Pull request by JamesWestb**

### CODE REVIEW

File: lib/discordbot/message_server_relay.ex [hunk 1]
```diff
@@ -49,10 +57,12 @@ defmodule Discordbot.MessageServerRelay do
 
     for guild_member <- guild_members do
```
Comment (plicjo) [inline lib/discordbot/message_server_relay.ex:58]: Typically, avoid using comprehensions in backend code (usually, I see them in templates) in favor of `Enum.map` or `Enum.each`

File: lib/discordbot/message_server_relay.ex [hunk 2]
```diff
@@ -49,10 +57,12 @@ defmodule Discordbot.MessageServerRelay do
...
       unless Enum.member?(employees, guild_member.user.username) do
```
Comment (plicjo) [inline lib/discordbot/message_server_relay.ex:59]: Lots of double negatives here. Let's just add a dilter here

```
Enum.filter do
  Enum.member?(employees, guild_member.user.username) && Enum.member?(guild_member.roles, System.get_env("ROLE_EMPLOYEE"))
end
```

This is not very performant code, but it's much easier to reason about. And with so few records, this is good to optimize readability over performance.

File: lib/discordbot/message_server_relay.ex [hunk 3]
```diff
@@ -29,10 +30,17 @@ defmodule Discordbot.MessageServerRelay do
     end
   end
 
-  def good_morning_message(dm_channel) do
-    message_channel("Good morning! What client did you work on yesterday?", dm_channel)
+  def good_morning_message() do
+    employees = employees()
 
-    PubSub.broadcast(Discordbot.PubSub, "#{dm_channel}", {:good_morning})
+    for employee <- employees do
```
Comment (plicjo) [inline lib/discordbot/message_server_relay.ex:36]: Prefer `Enum.map` over comprehensions.

**Review by plicjo — CHANGES_REQUESTED** [review]

File: lib/discordbot/message_server_relay.ex (hunk 2 above)
Comment (JamesWestb) [reply lib/discordbot/message_server_relay.ex:59]: My goal with this function is to ensure that new members entered into the database have not already been entered, and that they hold the correct Discord role. I understand though, the 'unless -> if' isn't very readable. I'll leave it at reformatting without the comprehension for now so I can re-deploy, and I'll see if I can find a way to address the readability tomorrow.

**Review by plicjo — CHANGES_REQUESTED** [review]
This works, but it’s still not idiomatic Elixir.

That means: “this looks like a Javascript dev wrote this.”

Lets pair up after stand up.

You should rarely be using if statements in Elixir. Instead, prefer pattern matching and piping.

We’re going to replace those ifs and unlesses with an Enum.filter and some piping.

File: lib/discordbot/message_server_relay.ex [hunk 4]
```diff
@@ -29,31 +30,48 @@ defmodule Discordbot.MessageServerRelay do
...
       @nostrum_api.list_guild_members(String.to_integer(System.get_env("COMPANY_SERVER")),
         limit: 30
       )
 
-    for guild_member <- guild_members do
-      unless Enum.member?(employees, guild_member.user.username) do
+    Enum.map(guild_members, fn guild_member -> update_employee(guild_member) end)
+  end
+
+  defp update_employee(guild_member) do
```
Comment (plicjo) [inline lib/discordbot/message_server_relay.ex:52]: I'm looking for something like this:

```
defp update_employee(guild_member) do
  Discordbot.Company.list_employees()
  |> Enum.map(fn list -> list.name end)
  |> Enum.filter(fn Enum.member?(employees, guild_member.user.username) && Enum.member?(guild_member.roles, 930_205_558_328_418_384))
  |> Enum.each(fn Discordbot.Company.create_employee(%{name: guild_member.user.username, user_id: Integer.to_string(guild_member.user.id) })
end
```

**Review by plicjo — CHANGES_REQUESTED** [review]

**Review by plicjo — CHANGES_REQUESTED** [review]

**Pull request merged**